volatility3/framework/plugins/linux/sockstat.py

```diff
@@ -0,0 +1,506 @@
+        """
+        if sock.has_member("sk_filter") and sock.sk_filter:
+            sock_filter = sock.sk_filter
+            extended["filter_type"] = "socket_filter"
+            self._extract_socket_filter_info(sock_filter, extended)
+
+        if sock.has_member("sk_reuseport_cb") and sock.sk_reuseport_cb:
+            sock_reuseport_cb = sock.sk_reuseport_cb
+            extended["filter_type"] = "reuseport_filter"
+            self._extract_socket_filter_info(sock_reuseport_cb, extended)
+
+    def _extract_socket_filter_info(self, sock_filter: objects.Pointer, extended: dict) -> None:
+        extended["bpf_filter_type"] = "cBPF"
+
+        if not sock_filter.has_member("prog") or not sock_filter.prog:
+            return
+
+        bpfprog = sock_filter.prog
+
+        # BPF_PROG_TYPE_UNSPEC = 0
+        if bpfprog.type > 0:
+            extended["bpf_filter_type"] = "eBPF"
+            bpfprog_aux = bpfprog.aux
+            if bpfprog_aux:
+                extended["bpf_filter_id"] = str(bpfprog_aux.id)
+                bpfprog_name = utility.array_to_string(bpfprog_aux.name)
+                if bpfprog_name:
+                    extended["bpf_filter_name"] = bpfprog_name
+
+    def _unix_sock(self, sock: objects.StructType) -> Tuple[objects.StructType, Tuple[str, str, str]]:
+        """Handles the AF_UNIX socket family
+
+        Args:
+            sock: Kernel generic `sock` object
+
+        Returns:
+            unix_sock: The kernel's `unix_sock` object
+            sock_stat: A tuple with the source, destination and state strings.
+        """
+        unix_sock = sock.cast("unix_sock")
+        state = unix_sock.get_state()
+        saddr = unix_sock.get_name()
+        sinode = unix_sock.get_inode()
+        if unix_sock.peer != 0:
+            peer = unix_sock.peer.dereference().cast("unix_sock")
+            daddr = peer.get_name()
+            dinode = peer.get_inode()
+        else:
+            daddr = dinode = ""
+
+        saddr_tag = f"{saddr} {sinode}"
+        daddr_tag = f"{daddr} {dinode}"
+        sock_stat = saddr_tag, daddr_tag, state
+        return unix_sock, sock_stat
+
+    def _inet_sock(self, sock: objects.StructType) -> Tuple[objects.StructType, Tuple[str, str, str]]:
+        """Handles the AF_INET/6 socket families
+
+        Args:
+            sock: Kernel generic `sock` object
+
+        Returns:
+            inet_sock: The kernel's `inet_sock` object
+            sock_stat: A tuple with the source, destination and state strings.
+        """
+        inet_sock = sock.cast("inet_sock")
+        saddr = inet_sock.get_src_addr()
+        sport = inet_sock.get_src_port()
+        daddr = inet_sock.get_dst_addr()
+        dport = inet_sock.get_dst_port()
+        state = inet_sock.get_state()
+
+        if inet_sock.get_family() == "AF_INET6":
+            saddr = f"[{saddr}]"
+
+        saddr_tag = f"{saddr}:{sport}"
+        daddr_tag = f"{daddr}:{dport}"
```

Comment:
Same as above, actually this one I think it's a little easier to understand with an unknown port number for example, still feels like something the code this deep shouldn't be deciding for everything above it (or at least document the behaviour in the docstring).